src/k8s-engine/images/nerdctlImageProcessor.ts

```diff
@@ -0,0 +1,140 @@
+   */
+  isInstallValid(mgr: K8s.KubernetesBackend, endpoint?: string): Promise<boolean> {
+    return new Promise(resolve => resolve(true));
+  }
+
+  /**
+   * Install the nerdctl backend should be a no-op
+   */
+  install(backend: K8s.KubernetesBackend, force = false, address?: string) {
+  }
+
+  async buildImage(dirPart: string, filePart: string, taggedImageName: string): Promise<imageProcessor.childResultType> {
+    const args = ['--namespace', this.currentNamespace, 'build',
+      '--file', path.join(dirPart, filePart),
+      '--tag', taggedImageName,
+      dirPart];
+
+    return await this.runImagesCommand(args);
+  }
+
+  async deleteImage(imageID: string): Promise<imageProcessor.childResultType> {
+    return await this.runImagesCommand(['--namespace', this.currentNamespace, 'rmi', imageID]);
+  }
+
+  async pullImage(taggedImageName: string): Promise<imageProcessor.childResultType> {
+    return await this.runImagesCommand(['--namespace', this.currentNamespace, 'pull', taggedImageName, '--debug']);
+  }
+
+  async pushImage(taggedImageName: string): Promise<imageProcessor.childResultType> {
+    return await this.runImagesCommand(['--namespace', this.currentNamespace, 'push', taggedImageName]);
+  }
+
+  async getImages(): Promise<imageProcessor.childResultType> {
+    return await this.runImagesCommand(['--namespace', this.currentNamespace, 'images'], false);
+  }
+
+  async scanImage(taggedImageName: string): Promise<imageProcessor.childResultType> {
+    return await this.runTrivyCommand(['image', '--no-progress', '--format', 'template',
+      '--template', '@/var/lib/trivy.tpl', taggedImageName]);
+  }
+
+  async getNamespaces(): Promise<Array<string>> {
+    const { stdout, stderr } = await childProcess.spawnFile(resources.executable('nerdctl'),
+      ['namespace', 'ls'],
+      { stdio: ['inherit', 'pipe', 'pipe'] });
+
+    if (stderr) {
+      console.log(`Error getting namespaces: ${ stderr }`, stderr);
+    }
+
+    return (stdout || '').toString().trimEnd().split(/\r?\n/).slice(1)
+      .map((line:string) => {
+        return line.split(/\s+/, 1)[0];
+      })
+      .sort();
+  }
+
+  /**
+   * Sample output:
+   * REPOSITORY        TAG       IMAGE ID        CREATED          SIZE
+   * camelpunch/pr     latest    f6b002c6f990    2 seconds ago    95.7 MiB
+   * ruby              latest    5139d3c9f2fc    9 minutes ago    911.0 MiB
+   * gibley/whoami     v01       31c94d15c40f    4 minutes ago    8.2 MiB
+   *                             31c94d15c40f    4 minutes ago    8.2 MiB
+
+   * @param data (like the above example)
+   *
+   * Because the input is so free-form, we assume it's tab-less ASCII, and parse the
+   * headers to determine the width of each column. Then those widths are used
+   * to parse each line. Duplicates are culled into the most informative line
+   */
+  parse(data: string): imageProcessor.imageType[] {
```

Comment:
Note for the future, nerdctl, docker, and podman all allow getting the images list as json. This may be easier than parsing string output in the future. Not saying we need to change that now.

Reply:
Worth changing now.